wire-ios-sync-engine/Tests/Source/Use cases/SetAllowGuestsAndServicesUseCaseTests.swift

```diff
@@ -0,0 +1,173 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModelSupport
+import XCTest
+
+@testable import WireSyncEngine
+
+final class SetAllowGuestsAndServicesUseCaseTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private let coreDataStackHelper = CoreDataStackHelper()
+    private var stack: CoreDataStack!
+    private let modelHelper = ModelHelper()
+    private var mockConversation: ZMConversation!
+    private var mockSelfUser: ZMUser!
+    private var sut: SetAllowGuestAndServicesUseCaseProtocol!
+
+    private var syncContext: NSManagedObjectContext {
+        return stack.syncContext
+    }
+
+    // MARK: - setUp
+
+    override func setUp() async throws {
+        try await super.setUp()
+        stack = try await coreDataStackHelper.createStack()
+        await syncContext.perform { [self] in
+            sut = SetAllowGuestAndServicesUseCase()
+            mockSelfUser = modelHelper.createSelfUser(in: syncContext)
+            mockConversation = modelHelper.createGroupConversation(in: syncContext)
+            mockConversation.teamRemoteIdentifier = UUID()
+        }
+    }
+
+    // MARK: - tearDown
+
+    override func tearDown() async throws {
+        stack = nil
+        sut = nil
+        try coreDataStackHelper.cleanupDirectory()
+        try await super.tearDown()
+    }
```

Comment:
Here the same, all properties should be released.

Reply:
Done!